core/templates/dev/head/exploration_editor/EditorServices.js

```diff
@@ -209,12 +243,35 @@ oppia.factory('changeListService', [
     gadget_visibility: true
   };
 
+  var autosaveChangeListOnChange = function(explorationChangeList) {
+    // Send autosave request and check for error in response:
```

Comment:
@jacobdavis11 Extracted this out into a separate function, and called this when discarding/undoing changes as well (until now, it was being called only when a change was being added!).